### PR TITLE
Add banner that suggest to switch to preferred locale by user

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #104924748, if applicable: 

## Summary

Add banner that suggest to switch to preferred locale by user

## Dependencies

NA

## Testing

Steps:
1. Add the following to `theme-settings.json` file at public folder
```json
{
  "features": {
    "docs": {
      "i18n": {
        "enable": true
      }
    }
  }
}
```
3. Run `npm run serve`
2. Go to the bottom of the page and change the language in the locale selector Chinese for example
3. Delete localStorage
4. If you have your browser in English, you should assert that a banner is displayed

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/LocaleSelector.vue [hunk 1]
```diff
@@ -48,9 +49,10 @@ export default {
     updateRouter({ target: { value: slug } }) {
       this.$router.push({
         params: {
-          locale: slug === defaultLocale ? null : slug,
+          locale: slug === defaultLocale ? undefined : slug,
```
Comment (dobromir-hristov) [inline src/components/LocaleSelector.vue:52]: why switch to `undefined`?

**Review by dobromir-hristov — COMMENTED** [review]
Code looks good. I cannot test, because I dont have a dataset, but make sure the extra header does not intervene with the sticky components, like the navigator, both on mobile and desktop variants.

I remember I added some guards against such things, but in any case...

**Comment by marinaaisa** [comment]
@swift-ci test

File: src/components/i18nBanner.vue [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      const appPreferredLocale = AppStore.state.preferredLocale;
+      // if user has a storaged preferred locale, return it
+      if (appPreferredLocale) return appPreferredLocale;
+      // find if user's navigator preference is available with the locales we provide
+      // in case it is, that will be the preferred locale to display
+      return locales.find((locale) => {
+        const languageAvailable = locale.code.split('-')[0];
+        const navigatorLanguage = window.navigator.language;
+        return navigatorLanguage.includes(languageAvailable);
+      }).slug;
```
Comment (dobromir-hristov) [inline src/components/i18nBanner.vue:58]: if `find` returns `undefined` this will throw.

File: src/components/i18nBanner.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    preferredLocale: () => {
+      const appPreferredLocale = AppStore.state.preferredLocale;
+      // if user has a storaged preferred locale, return it
+      if (appPreferredLocale) return appPreferredLocale;
+      // find if user's navigator preference is available with the locales we provide
+      // in case it is, that will be the preferred locale to display
+      return locales.find((locale) => {
+        const languageAvailable = locale.code.split('-')[0];
+        const navigatorLanguage = window.navigator.language;
+        return navigatorLanguage.includes(languageAvailable);
```
Comment (dobromir-hristov) [inline src/components/i18nBanner.vue:57]: I wonder if `includes` could return a false positive... we are comparing `en` to `en` in most cases, right?

File: src/components/LocaleSelector.vue (hunk 1 above)
Comment (dobromir-hristov) [reply src/components/LocaleSelector.vue:52]: its redirecting for me? I just tried 🤔

File: src/components/i18nBanner.vue (hunk 3 above)
Comment (marinaaisa) [reply src/components/i18nBanner.vue]: true, I'm comparing the languages now: https://github.com/apple/swift-docc-render/pull/554/commits/20363e62b74e680959ce486e7e1ce24a3ee2469c

**Comment by marinaaisa** [comment]
Thanks Dobri! I've addressed your feedback. Give it another try :)

**Review by dobromir-hristov — COMMENTED** [review]
LGTM, will wait on design to approve too

File: src/App.vue [hunk 4]
```diff
@@ -17,6 +17,7 @@
     <a href="#main" id="skip-nav">{{ $t('accessibility.skip-navigation') }}</a>
     <InitialLoadingPlaceholder />
     <slot name="header" :isTargetIDE="isTargetIDE">
+      <SuggestLang v-if="enablei18n" />
```
Comment (dobromir-hristov) [inline src/App.vue:20]: Why not add it above the slot 🤔

**Review by dobromir-hristov — APPROVED** [review]

File: src/components/SuggestLang.vue [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+@import 'docc-render/styles/_core.scss';
+
+$banner-icon-padding: $nav-padding-small / 4;
+
+.suggest-lang {
+  background: dark-color(fill);
+  color: light-color(fill);
+  display: flex;
+  justify-content: center;
+  border-bottom: 1px solid var(--color-grid);
```
[the remark this answers is not in the document]
Comment (marinaaisa) [reply src/components/SuggestLang.vue:85]: Yes, the dark theme has the same background color for the SuggestLang and rest of the page, so it's difficult to see the boundaries. We use this border for other parts of the UI that are next to the main content (footer, navigator, asides...), so I used it here in the same way.

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test

**Review by dobromir-hristov — APPROVED** [review]

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test

**Pull request merged**